### ISSUE TITLE
On leaf-of-tree resource which is down, imv / irm succeed with error

### ISSUE THREAD

**Issue by mcast** [irods/irods#5487]
## Bug Report
Thanks for compiling the table in  #4508 - this issue is covers a few cases from that general issue: for leaf resource, `imv`, `itrim` and `irm`. These are operations we do more often, and could be blocked by resources being down.

### iRODS Version, OS and Version
iRODS 4.2.7 on Ubuntu Bionic (server & client)

### What did you try to do?
Full transcript at the bottom (including some faffing around), but essentially
1. make a file with replicas in "green" and "red" trees
2. mark the "red" replica down; but the service is still running
3. try commands and look at results
    * `imv`: error but success
    * `itrim`: error and fail (seems to be correct)
    * `irm`: error but success
4. did not look at actual file vault state during; for me this is possible but not convenient with our current dev zone
5. further complication: when a resource was really down / off the network / no power,
    * we might still want to `imv`, `itrim` or `irm` with actual effect
    * it would suit us to have the action upon the vault queued up for later effect, but leaving cruft in the vault won't be new and the system currently deals with those when it has to
    * I realise this last bit is a feature request inside a bug report 😁 , do you want another issue for that?

### Expected behavior
* If there is an error, the operation should not have happened
* If the operation succeeded, don't raise an error
* If the operation did not follow these rules, the error message should indicate that the file may be in an unexpected state

### Observed behavior (including steps to reproduce, if applicable)
Error messages and exit codes can't be trusted. In our work pattern this is merely confusing, because we don't say `irm` and then lament the file's loss!

```
mca@casm3-head1:~$ i-cgpdev iadmin modresc irods-cgp-dev-e01-sdb status down
mca@casm3-head1:~$ i-cgpdev ilsresc-up-the-tree 
demoResc:unixfilesystem                            (up)
green:passthru                                     (up)
`-- greenrandom:random                             (up)
    `-- green1:random                              (up)
        |-- irods-cgp-sr15-dev-sda:unixfilesystem  up
        |-- irods-cgp-sr15-dev-sdb:unixfilesystem  up
        |-- irods-cgp-sr15-dev-sdc:unixfilesystem  up
        |-- irods-cgp-sr15-dev-sdd:unixfilesystem  up
        `-- irods-cgp-sr15-dev-sdf:unixfilesystem  up
irods-cgp-dev-e02Resource:unixfilesystem           (up)
irods-cgp-sr15-devResource:unixfilesystem          up
purple:passthru                                    (up)
`-- purplerandom:random                            (up)
    `-- purple1:random                             (up)
        `-- irods-cgp-sr15-dev-sde:unixfilesystem  up
red:passthru                                       (up)
`-- redrandom:random                               (up)
    `-- red1:random                                (up)
        |-- irods-cgp-dev-e01-sdb:unixfilesystem   down
        |-- irods-cgp-dev-e01-sdc:unixfilesystem   (up)
        |-- irods-cgp-dev-e02-sdb:unixfilesystem   (up)
        `-- irods-cgp-dev-e02-sdc:unixfilesystem   (up)
mca@casm3-head1:~$ i-cgpdev ils -L junk
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
mca@casm3-head1:~$ i-cgpdev imv junk junk~
remote addresses: 172.30.157.15 ERROR: mvUtil: mv error for /cgp_dev/sandbox/mca/junk~, status = -818000 status = -818000 CAT_NO_ACCESS_PERMISSION
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
  C- /cgp_dev/sandbox/mca/dev1
mca@casm3-head1:~$ i-cgpdev itrim -N1 -n1 junk
remote addresses: 172.30.157.15 ERROR: trimUtil: trim error for /cgp_dev/sandbox/mca/junk.  status = -1803000 HIERARCHY_ERROR
Total size trimmed = 0.000 MB. Number of files trimmed = 0.
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
  C- /cgp_dev/sandbox/mca/dev1
mca@casm3-head1:~$ i-cgpdev irm junk 
remote addresses: 172.30.157.15 ERROR: rmUtil: rm error for /cgp_dev/sandbox/mca/junk, status = -818000 status = -818000 CAT_NO_ACCESS_PERMISSION
mca@casm3-head1:~$ i-cgpdev iuserinfo
name: mca2
id: 13040
type: rodsadmin
zone: cgp_dev
info: 
comment: secondary user with non-LDAP native-iinit authentication, in case I break the mca(LDAP) user
create time: 01592820357: 2020-06-22.11:05:57
modify time: 01592820425: 2020-06-22.11:07:05
member of group: mca2
member of group: public
mca@casm3-head1:~$ clientUserName=mca i-cgpdev ichmod junk own mca2
remote addresses: 172.30.157.15 ERROR: rcModAccessControl failure  status = -816000 CAT_INVALID_ARGUMENT
remote addresses: 172.30.157.15 ERROR: Level 0: access level 'junk' is invalid
Level 0: access level 'junk' is invalid
mca@casm3-head1:~$ clientUserName=mca i-cgpdev ichmod own mca2 junk 
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
  C- /cgp_dev/sandbox/mca/dev1
mca@casm3-head1:~$ i-cgpdev imv junk junk~
remote addresses: 172.30.157.15 ERROR: mvUtil: mv error for /cgp_dev/sandbox/mca/junk~, status = -105000 status = -105000 SYS_RESC_IS_DOWN
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk~
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk~
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk~
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
  C- /cgp_dev/sandbox/mca/dev1
mca@casm3-head1:~$ i-cgpdev imv junk~ junk~~ ; echo $?
remote addresses: 172.30.157.15 ERROR: mvUtil: mv error for /cgp_dev/sandbox/mca/junk~~, status = -105000 status = -105000 SYS_RESC_IS_DOWN
3
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk~~
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk~~
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk~~
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
  C- /cgp_dev/sandbox/mca/dev1
mca@casm3-head1:~$ i-cgpdev itrim -n1 -N1 junk~~ 
remote addresses: 172.30.157.15 ERROR: trimUtil: trim error for /cgp_dev/sandbox/mca/junk~~.  status = -1803000 HIERARCHY_ERROR
Total size trimmed = 0.000 MB. Number of files trimmed = 0.
mca@casm3-head1:~$ echo $?
3
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sdc         1667 2020-07-07.14:56 & junk~~
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-sr15-dev-sdc/sandbox/mca/junk~~
  mca               1 red;redrandom;red1;irods-cgp-dev-e01-sdb         1667 2021-03-15.10:39 & junk~~
    29f89a9a53c9c3b820d078aa6e9d9b3d    generic    /irods-cgp-dev-e01-sdb/sandbox/mca/junk
  C- /cgp_dev/sandbox/mca/dev1
mca@casm3-head1:~$ i-cgpdev irm junk~~ ; echo $?
remote addresses: 172.30.157.15 ERROR: rmUtil: rm error for /cgp_dev/sandbox/mca/junk~~, status = -105000 status = -105000 SYS_RESC_IS_DOWN
Level 0: collection '/cgp_dev/trash/orphan/mca2#cgp_dev' is unknown
3
mca@casm3-head1:~$ i-cgpdev ils -L 
/cgp_dev/sandbox/mca:
  mca               0 green;greenrandom;green1;irods-cgp-sr15-dev-sda         3118 2020-04-07.16:57 & cgp-zone.txt
    7147b93153fa1d049f546213a4c78dbc    generic    /irods-cgp-sr15-dev-sda/sandbox/mca/cgp-zone.txt
  C- /cgp_dev/sandbox/mca/dev1
```

`i-cgpdev` is a prefix command which runs `"$@"` after exporting `IRODS_ENVIRONMENT_FILE`

**Comment by mcast** [comment]
Can I draw attention to the distinction "marked down" vs "actually down"?

With current dev zone, I can only mark the resource down. I can't actually stop the iRES, remove its vault filesystem or `mount -o remount,ro ...` that filesystem.

**Comment by trel** [comment]
I marked up your original examples a little for clarity/separation.

------

Okay, so I see...

1) Without permissions... nothing happened to the targeted data object.   And I think this is expected and good.  The error messages are a bit all over the place (some already fixed/better in 4-2-stable), but the catalog and the data both seem to be okay/sane in this scenario.

2) With permissions...

 - `imv` moved both replicas in the logical namespace, but only moved replica 0's physical file (replica 1's resource is marked `down`).  It returned an error code.  I agree, this should be either a warning with the current behavior (logical name *was* moved), OR, this should unwind the `imv` altogether.   I think unwinding is probably too much -- something being marked down is not necessarily an error, it could be a bit of planned maintenance... or the disk is truly already known to be dead and will not be coming back.

 - `itrim` failed to trim (and returned an error) b/c the physical medium could not do the requested job (unlink a file) - so this seems fine.

 - `irm` without the `-f` is a logical operation (and equivalent to an `imv` to the trash)...  In this case, a special case of moving it to the orphan area of the trash due to the earlier `imv` that could not touch replica 1's physical file (which, completely independently, failed). 

---

I'm not sure about your comment regarding 'marked down' vs 'actually down'.  The decision-making that the iRODS server needs to do does not care about that distinction.   If the resource is marked down, then before any actions are taken, 'being down' is taken into consideration during 'voting'.  If the resource is marked up, but is physically not present (aka NFS has a hiccup or someone actually unmounts a Vault's drive) or doesn't have the right unix permissions... then the voting would still route the operation to the physical layer... and then a physical read/write/permission error would occur.

All that said, the reason that #4508 is still open is because there are numerous (competing?) scenarios where it's not (yet) exactly clear what the behavior *should* be.   As we write tests and exercise these edge cases, we'll develop some better heuristics to help us reason it all.

Is there anything actionable here that isn't covered by #4508?

**Comment by mcast** [comment]
(With apologies for the delay,) I ran `irm` and `imv` cases while watching more closely what happens. We have an NFS export of the `unixfilesystemvault` behind the `irods-cgp-sr15dev-*` resources which makes this easy.

TLDR; I'm now less inclined to call this behaviour a bug, but I'm sure I've not read any documentation describing what will happen.

## `irm`

With `irm` when a resource is down, the file disappears from the iCAT and resources which are up.  The physical file in the `status:down` vault remains after, and would be displaced if a subsequent `iput` landed in the same place.

Cruft is left behind which ought to be deleted - for tidiness, for space recovery and for GDPR.

```
$ i-cgpdev ils -L /cgp_dev/home/mca2/foo
  mca2              0 green;greenrandom;green1;irods-cgp-sr15-dev-sdd         4047 2021-11-23.17:28 & foo
    247719b01c0508bc8fbfeb18c12c30ca    generic    /irods-cgp-sr15-dev-sdd/home/mca2/foo
  mca2              1 red;redrandom;red1;irods-cgp-dev-e02-sdc         4047 2021-11-25.15:39 & foo
    247719b01c0508bc8fbfeb18c12c30ca    generic    /irods-cgp-dev-e02-sdc/home/mca2/foo
  mca2              2 purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc         4047 2021-11-25.15:39 & foo
    247719b01c0508bc8fbfeb18c12c30ca    generic    /irods-cgp-sr15-dev-sdc/home/mca2/foo
$ i-cgpdev itrim -N2 -n0 /cgp_dev/home/mca2/foo; echo $?
remote addresses: 172.30.157.15 ERROR: trimUtil: trim error for /cgp_dev/home/mca2/foo.  status = -1803000 HIERARCHY_ERROR
Total size trimmed = 0.000 MB. Number of files trimmed = 0.
3
$ i-cgpdev ils -L /cgp_dev/home/mca2/foo
  mca2              0 green;greenrandom;green1;irods-cgp-sr15-dev-sdd         4047 2021-11-23.17:28 & foo
    247719b01c0508bc8fbfeb18c12c30ca    generic    /irods-cgp-sr15-dev-sdd/home/mca2/foo
  mca2              1 red;redrandom;red1;irods-cgp-dev-e02-sdc         4047 2021-11-25.15:39 & foo
    247719b01c0508bc8fbfeb18c12c30ca    generic    /irods-cgp-dev-e02-sdc/home/mca2/foo
  mca2              2 purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc         4047 2021-11-25.15:39 & foo
    247719b01c0508bc8fbfeb18c12c30ca    generic    /irods-cgp-sr15-dev-sdc/home/mca2/foo
$ ls -l /nfs/irods-cgp-sr15-dev-sd*/home/mca2/foo
-rw-r----- 1 irods-cgp cancer 4047 Nov 25 15:39 /nfs/irods-cgp-sr15-dev-sdc/home/mca2/foo
-rw-r----- 1 irods-cgp cancer 4047 Nov 23 17:28 /nfs/irods-cgp-sr15-dev-sdd/home/mca2/foo
$ i-cgpdev irm /cgp_dev/home/mca2/foo; echo $?
remote addresses: 172.30.157.15 ERROR: rmUtil: rm error for /cgp_dev/home/mca2/foo, status = -105000 status = -105000 SYS_RESC_IS_DOWN
Level 0: collection '/cgp_dev/trash/orphan/mca2#cgp_dev' is unknown
3
$ ls -l /nfs/irods-cgp-sr15-dev-sd*/home/mca2/foo
-rw-r----- 1 irods-cgp cancer 4047 Nov 23 17:28 /nfs/irods-cgp-sr15-dev-sdd/home/mca2/foo
$ i-cgpdev iadmin modresc irods-cgp-sr15-dev-sdd status up
```

## `imv`

The `imv` is more complex, and tracks state in a way I didn't expect.

```
$ i-cgpdev iput *.fofn0.bz2 /cgp_dev/home/mca2/bar
$ i-cgpdev irepl -R red /cgp_dev/home/mca2/bar
$ i-cgpdev irepl -R purple /cgp_dev/home/mca2/bar
$ i-cgpdev ils -L /cgp_dev/home/mca2/bar
  mca2              0 green;greenrandom;green1;irods-cgp-sr15-dev-sdb      2358709 2021-11-25.15:46 & bar
    cf6b798109b368161c685283d9b24d02    generic    /irods-cgp-sr15-dev-sdb/home/mca2/bar
  mca2              1 red;redrandom;red1;irods-cgp-dev-e01-sdb      2358709 2021-11-25.15:47 & bar
    cf6b798109b368161c685283d9b24d02    generic    /irods-cgp-dev-e01-sdb/home/mca2/bar
  mca2              2 purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc      2358709 2021-11-25.15:47 & bar
    cf6b798109b368161c685283d9b24d02    generic    /irods-cgp-sr15-dev-sdc/home/mca2/bar
$ i-cgpdev iadmin modresc irods-cgp-sr15-dev-sdb status down
$ ls -l /nfs/irods-cgp-sr15-dev-sd*/home/mca2/bar
-rw-r----- 1 irods-cgp cancer 2358709 Nov 25 15:46 /nfs/irods-cgp-sr15-dev-sdb/home/mca2/bar
-rw-r----- 1 irods-cgp cancer 2358709 Nov 25 15:47 /nfs/irods-cgp-sr15-dev-sdc/home/mca2/bar
$ i-cgpdev imv /cgp_dev/home/mca2/{bar,baz}
remote addresses: 172.30.157.15 ERROR: mvUtil: mv error for /cgp_dev/home/mca2/baz, status = -105000 status = -105000 SYS_RESC_IS_DOWN
$ echo $?
3
$ i-cgpdev ils -L /cgp_dev/home/mca2/bar
remote addresses: 172.30.157.15 ERROR: lsUtil: srcPath /cgp_dev/home/mca2/bar does not exist or user lacks access permission
$ i-cgpdev ils -L /cgp_dev/home/mca2/baz
  mca2              0 green;greenrandom;green1;irods-cgp-sr15-dev-sdb      2358709 2021-11-25.15:46 & baz
    cf6b798109b368161c685283d9b24d02    generic    /irods-cgp-sr15-dev-sdb/home/mca2/bar
  mca2              1 red;redrandom;red1;irods-cgp-dev-e01-sdb      2358709 2021-11-25.15:47 & baz
    cf6b798109b368161c685283d9b24d02    generic    /irods-cgp-dev-e01-sdb/home/mca2/baz
  mca2              2 purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc      2358709 2021-11-25.15:47 & baz
    cf6b798109b368161c685283d9b24d02    generic    /irods-cgp-sr15-dev-sdc/home/mca2/baz
```

The file is moved while one copy is unavailable.

```
$ ls -l /nfs/irods-cgp-sr15-dev-sd*/home/mca2/ba?
-rw-r----- 1 irods-cgp cancer 2358709 Nov 25 15:46 /nfs/irods-cgp-sr15-dev-sdb/home/mca2/bar
-rw-r----- 1 irods-cgp cancer 2358709 Nov 25 15:47 /nfs/irods-cgp-sr15-dev-sdc/home/mca2/baz
$ i-cgpdev iget -n0 /cgp_dev/home/mca2/baz baz0
remote addresses: 172.30.157.15 ERROR: getUtil: get error for baz0 status = -1803000 HIERARCHY_ERROR
$ i-cgpdev iadmin modresc irods-cgp-sr15-dev-sdb status up
$ i-cgpdev iget -n0 /cgp_dev/home/mca2/baz baz0
$ ls -l baz0
-rw-r----- 1 mca cancer 2358709 Nov 25 15:50 baz0
$ ls -l /nfs/irods-cgp-sr15-dev-sd*/home/mca2/ba?
-rw-r----- 1 irods-cgp cancer 2358709 Nov 25 15:46 /nfs/irods-cgp-sr15-dev-sdb/home/mca2/bar
-rw-r----- 1 irods-cgp cancer 2358709 Nov 25 15:47 /nfs/irods-cgp-sr15-dev-sdc/home/mca2/baz
$ i-cgpdev iquest '%s %s' "select DATA_RESC_HIER, DATA_PATH where DATA_NAME = 'baz'"
green;greenrandom;green1;irods-cgp-sr15-dev-sdb /irods-cgp-sr15-dev-sdb/home/mca2/bar
purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc /irods-cgp-sr15-dev-sdc/home/mca2/baz
red;redrandom;red1;irods-cgp-dev-e01-sdb /irods-cgp-dev-e01-sdb/home/mca2/baz
```

Well I didn't expect that!

What about moves to another collection? That follows the same pattern, and eventually catches up if allowed to.

```
$ i-cgpdev iadmin modresc irods-cgp-sr15-dev-sdb status down
$ i-cgpdev imkdir /cgp_dev/home/mca2/spork
$ i-cgpdev imv /cgp_dev/home/mca2/baz /cgp_dev/home/mca2/spork/baz 
remote addresses: 172.30.157.15 ERROR: mvUtil: mv error for /cgp_dev/home/mca2/spork/baz, status = -105000 status = -105000 SYS_RESC_IS_DOWN
$ i-cgpdev iquest '%s %s' "select DATA_RESC_HIER, DATA_PATH where DATA_NAME = 'baz'"
green;greenrandom;green1;irods-cgp-sr15-dev-sdb /irods-cgp-sr15-dev-sdb/home/mca2/bar
purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc /irods-cgp-sr15-dev-sdc/home/mca2/spork/baz
red;redrandom;red1;irods-cgp-dev-e01-sdb /irods-cgp-dev-e01-sdb/home/mca2/spork/baz
$ i-cgpdev iadmin modresc irods-cgp-sr15-dev-sdb status up
$ i-cgpdev iget -n0 /cgp_dev/home/mca2/baz - | md5sum
remote addresses: 172.30.157.15 ERROR: resolveRodsTarget: srcPath /cgp_dev/home/mca2/baz does not exist
remote addresses: 172.30.157.15 ERROR: getUtil: resolveRodsTarget status = -317000 USER_INPUT_PATH_ERR
d41d8cd98f00b204e9800998ecf8427e  -
$ i-cgpdev iget -n0 /cgp_dev/home/mca2/spork/baz - | md5sum
cf6b798109b368161c685283d9b24d02  -
$ i-cgpdev iget -n1 /cgp_dev/home/mca2/spork/baz - | md5sum
cf6b798109b368161c685283d9b24d02  -
$ i-cgpdev imv /cgp_dev/home/mca2/spork/baz /cgp_dev/home/mca2/spork/zabbit
$ i-cgpdev iquest '%s %s' "select DATA_RESC_HIER, DATA_PATH where DATA_NAME = 'zabbit'"
green;greenrandom;green1;irods-cgp-sr15-dev-sdb /irods-cgp-sr15-dev-sdb/home/mca2/spork/zabbit
purple;purplerandom;purple1;irods-cgp-sr15-dev-sdc /irods-cgp-sr15-dev-sdc/home/mca2/spork/zabbit
red;redrandom;red1;irods-cgp-dev-e01-sdb /irods-cgp-dev-e01-sdb/home/mca2/spork/zabbit
```
